### PR TITLE
Remove deprecated rpc options

### PR DESCRIPTION
**Pull request by jnewbery**
There were some RPC/RPC options deprecated in v0.16. Those can now be removed from master since v0.16 has been branched.

- `estimatefee` RPC has been removed. The `feature_fee_estimation.py` test has been updated to remove the RPC, but doesn't yet have good coverage of the replacement RPC `estimatesmartfee`. Improving the test coverage should be done in a new PR. (#11031)
- the `errors` field returned by `getmininginfo` has been deprecated and replaced by a `warning` field. (#10858)
- providing addresses as inputs to `createmultisig` has been deprecated. Users should use `addmultisigaddress` instead (#11415)
- The return format from `addmultisigaddress` has changed (#11415)

`getwitnessaddress` was also deprecated in v0.16 and can be removed, but many tests are using that RPC, so it's a larger job to remove. It should be removed in a separate PR (possibly after #11739 and #11398 have been merged and the segwit test code tidied up)

### CODE REVIEW

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -772,43 +767,8 @@ UniValue submitblock(const JSONRPCRequest& request)
...
-    int nBlocks = request.params[0].get_int();
-    if (nBlocks < 1)
-        nBlocks = 1;
-
-    CFeeRate feeRate = ::feeEstimator.estimateFee(nBlocks);
-    if (feeRate == CFeeRate(0))
-        return -1.0;
-
-    return ValueFromAmount(feeRate.GetFeePerK());
+    throw JSONRPCError(RPC_METHOD_DEPRECATED, "estimatefee was removed in v0.17.\n"
```
Comment (promag) [inline src/rpc/mining.cpp]: Why not remove?

File: src/rpc/mining.cpp [hunk 2]
```diff
@@ -219,11 +218,7 @@ UniValue getmininginfo(const JSONRPCRequest& request)
     obj.push_back(Pair("networkhashps",    getnetworkhashps(request)));
     obj.push_back(Pair("pooledtx",         (uint64_t)mempool.size()));
     obj.push_back(Pair("chain",            Params().NetworkIDString()));
-    if (IsDeprecatedRPCEnabled("getmininginfo")) {
-        obj.push_back(Pair("errors",       GetWarnings("statusbar")));
-    } else {
-        obj.push_back(Pair("warnings",     GetWarnings("statusbar")));
-    }
+    obj.push_back(Pair("warnings",     GetWarnings("statusbar")));
```
Comment (promag) [inline src/rpc/mining.cpp]: Nit, to align or not to align 👀 IMO one space.

Comment (jnewbery) [reply src/rpc/mining.cpp]: Fine. Done.

File: src/rpc/mining.cpp (hunk 1 above)
Comment (jnewbery) [reply src/rpc/mining.cpp]: Because there's no harm in leaving the warning in for one additional release. Output is now:

```
→ bcli estimatefee 4
error code: -32
error message:
estimatefee was removed in v0.17.
Clients should use estimatesmartfee.
```

Comment (promag) [reply src/rpc/mining.cpp]: Alright, in the future we should do the same for others. Not sure if this is documented (or should be).

File: src/rpc/mining.cpp [hunk 3]
```diff
@@ -986,7 +946,7 @@ static const CRPCCommand commands[] =
 
     { "generating",         "generatetoaddress",      &generatetoaddress,      {"nblocks","address","maxtries"} },
 
-    { "util",               "estimatefee",            &estimatefee,            {"nblocks"} },
+    { "util",               "estimatefee",            &estimatefee,            {} },
```
Comment (MarcoFalke) [inline src/rpc/mining.cpp]: Should be hidden, I guess.

Comment (jnewbery) [reply src/rpc/mining.cpp]: Yes. Fixed.

**Comment by sipa** [comment]
Concept ACK

File: test/functional/rpc_deprecated.py [hunk 4]
```diff
@@ -4,24 +4,15 @@
...
-        self.log.info("estimatefee: Shows deprecated message")
-        assert_raises_rpc_error(-32, 'estimatefee is deprecated', self.nodes[0].estimatefee, 1)
-
-        self.log.info("Using -deprecatedrpc=estimatefee bypasses the error")
-        self.nodes[1].estimatefee(1)
-
-        self.log.info("Make sure that -deprecatedrpc=createmultisig allows it to take addresses")
-        assert_raises_rpc_error(-5, "Invalid public key", self.nodes[0].createmultisig, 1, [self.nodes[0].getnewaddress()])
-        self.nodes[1].createmultisig(1, [self.nodes[1].getnewaddress()])
+        pass
```
Comment (laanwj) [inline test/functional/rpc_deprecated.py]: Please add a comment here that newly deprecated functionality should be tested here, otherwise this empty function looks really strange.
Alternatively, maybe it's better to remove this test entirely if it's empty, It can always be brought back.

File: src/rpc/mining.cpp (hunk 1 above)
Comment (laanwj) [reply src/rpc/mining.cpp]: Nit: No need for \n at the end of exception messages.

**Review by laanwj — COMMENTED** [review]
Looks good to me, a few small nits.
Would make sense to mention final removal of these in `doc/release-notes.md` (the one on the master branch).

File: test/functional/rpc_deprecated.py (hunk 4 above)
Comment (promag) [reply test/functional/rpc_deprecated.py]: IMO should stay otherwise it's easy to add deprecated tests elsewhere. A comment would be nice.

File: src/rpc/misc.cpp [hunk 5]
```diff
@@ -297,15 +294,8 @@ UniValue createmultisig(const JSONRPCRequest& request)
...
-#ifdef ENABLE_WALLET
-            CWallet* const pwallet = GetWalletForJSONRPCRequest(request);
-            if (IsDeprecatedRPCEnabled("createmultisig") && EnsureWalletIsAvailable(pwallet, false)) {
-                pubkeys.push_back(AddrToPubKey(pwallet, keys[i].get_str()));
-            } else
-#endif
             throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Invalid public key: %s\nNote that from v0.16, createmultisig no longer accepts addresses."
-            " Clients must transition to using addmultisigaddress to create multisig addresses with addresses known to the wallet before upgrading to v0.17."
-            " To use the deprecated functionality, start bitcoind with -deprecatedrpc=createmultisig", keys[i].get_str()));
+            " Users must use addmultisigaddress to create multisig addresses with addresses known to the wallet.", keys[i].get_str()));
```
Comment (promag) [inline src/rpc/misc.cpp]: I believe there is no test for this error?

File: src/rpc/mining.cpp (hunk 1 above)
Comment (promag) [reply src/rpc/mining.cpp]: I guess there is no need for `\n`. There are other multiple sentence errors without it.

**Review by promag — COMMENTED** [review]
Tested ACK.

> Would make sense to mention final removal of these in `doc/release-notes.md`

Or needs release notes tag.

I would reword commit "[tests] Remove tests for deprecated estimatefee RPC" to "[qa] Replace deprecated estimatefee with estimatesmartfee".

Also remove:?
https://github.com/bitcoin/bitcoin/blob/c9ca4f6024e01fdca509ae07887e6fe2157c6384/src/wallet/rpcwallet.cpp#L3140

File: test/functional/rpc_deprecated.py (hunk 4 above)
Comment (jnewbery) [reply test/functional/rpc_deprecated.py]: Comment added.

File: src/rpc/mining.cpp (hunk 1 above)
Comment (jnewbery) [reply src/rpc/mining.cpp]: ugh. This error message was split over two lines before the commit. Please can we focus on reviewing the code changes and not how many new lines there should be in error messages?

File: src/rpc/misc.cpp (hunk 5 above)
Comment (jnewbery) [reply src/rpc/misc.cpp]: https://github.com/bitcoin/bitcoin/blob/b2645286745b77a8268d00497e0cbccecf42952c/test/functional/rpc_rawtransaction.py#L155

**Comment by jnewbery** [comment]
> I would reword commit "[tests] Remove tests for deprecated estimatefee RPC" to "[qa] Replace deprecated estimatefee with estimatesmartfee".

That's not what the commit does. It removes the tests for estimatefee and leaves the (minimal) checking of estimatesmartfee in place.

Please see original comment:

> The `feature_fee_estimation.py` test has been updated to remove the RPC, but doesn't yet have good coverage of the replacement RPC estimatesmartfee. Improving the test coverage should be done in a new PR.

Feel free to contribute better testing for estimatesmartfee.

File: test/functional/rpc_deprecated.py [hunk 6]
```diff
@@ -4,24 +4,24 @@
...
-        self.nodes[1].createmultisig(1, [self.nodes[1].getnewaddress()])
+        # This test should be used to verify correct behaviour of deprecated
+        # RPC methods with and without the -deprecatedrpc flags. For example:
+        #
+        # self.log.info("Make sure that -deprecatedrpc=createmultisig allows it to take addresses")
+        # assert_raises_rpc_error(-5, "Invalid public key", self.nodes[0].createmultisig, 1, [self.nodes[0].getnewaddress()])
+        # self.nodes[1].createmultisig(1, [self.nodes[1].getnewaddress()])
+        #
+        # There are currently no deprecated RPC methods in master, so this
+        # test is currently empty.
```
Comment (laanwj) [inline test/functional/rpc_deprecated.py]: Thanks
